components/director/pkg/graphql/app_template_validation.go

```diff
@@ -69,6 +69,62 @@ func (i ApplicationTemplateInput) ensurePlaceholdersUsed() error {
 	return nil
 }
 
+func (i ApplicationTemplateUpdateInput) Validate() error {
+	return validation.Errors{
+		"Rule.ValidPlaceholders": i.validPlaceholders(),
+		"name":                   validation.Validate(i.Name, validation.Required, is.PrintableASCII, validation.Length(1, 100)),
+		"description":            validation.Validate(i.Description, validation.RuneLength(0, descriptionStringLengthLimit)),
+		"placeholders":           validation.Validate(i.Placeholders, validation.Each(validation.Required)),
+		"accessLevel":            validation.Validate(i.AccessLevel, validation.Required, validation.In(ApplicationTemplateAccessLevelGlobal)),
+	}.Filter()
+}
+
+func (i ApplicationTemplateUpdateInput) validPlaceholders() error {
+	if err := i.ensureUniquePlaceholders(); err != nil {
+		return err
+	}
+	if err := i.ensurePlaceholdersUsed(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i ApplicationTemplateUpdateInput) ensureUniquePlaceholders() error {
```

Comment:
several methods here are present twice in the same file.. thats too much duplication